app/assets/javascripts/angular/controllers/topics_ctrl.coffee

```diff
@@ -15,6 +15,15 @@ angular.module "lean-coffee"
       if confirm('Are you sure you want to delete this?')
         topicsService.delete(topic)
 
+    @archive = (topic) =>
+      if confirm('Are you sure you want to archive this?')
+        topicsService.archive(topic)
+
+    @archiveAll = =>
+      if confirm('Are you sure you want to archive all discussed topics?')
+        topicsService.archiveAll()
+        topicsService.reload()
```

Comment:
`archive` works on one topic. Like `update` and `delete`, once that topic is updated, Angular (somehow) knows to update the data it's holding onto, and its two-way data binding causes the card for that topic to get refreshed (in the case of `delete`, it causes it to be removed).

That "somehow" is a mystery I didn't fully probe. I'm not sure if Angular listens for the return value from the server, or if it gets the message from Pusher. Maybe whichever one is first. It's possible that all `TopicsController` json responses could be empty.

In any case, [`topicsService.topics` is being fetched at page load and then never updated](https://github.com/chadoh/lean-coffee/blob/rooms/app/assets/javascripts/angular/services/topics_service.coffee#L3). 

You know, now that I'm thinking about this, I don't think `archiveAll` is going to get pushed to all clients correctly. I am in fact [publishing about each one that was updated](https://github.com/chadoh/lean-coffee/blob/rooms/app/controllers/topics_controller.rb#L47), but [the Angular template I wrote](https://github.com/chadoh/lean-coffee/blob/rooms/app/views/rooms/show.html.slim#L27) wasn't dealing with it properly and hiding them, and my crappy solution at the time was just to reload all topics. But that will only work for the browser making the request. I shouldn't need this `reload`. I'll improve this now.

Thanks for asking about it!